src/SDK/Resource/ResourceInfo.php

```diff
@@ -42,15 +45,15 @@ public static function create(AttributesInterface $attributes, ?string $schemaUr
      */
     public static function merge(ResourceInfo ...$resources): self
     {
-        $attributes = [];
-        $schemaUrl = null;
-
-        foreach ($resources as $resource) {
-            $attributes += $resource->getAttributes()->toArray();
-            $schemaUrl ??= $resource->getSchemaUrl();
+        if (empty($resources)) {
+            return new ResourceInfo(new Attributes([]), null);
         }
 
-        return new ResourceInfo(new Attributes($attributes), $schemaUrl);
+        $initialResource = array_shift($resources);
+
+        return array_reduce($resources, function (ResourceInfo $old, ResourceInfo $updating): ResourceInfo {
+            return $old->mergeWith($updating);
```

Comment:
Calling the `mergeWith` will result in unnecessary object creation when used in `defaultResource`, however this could be optomized later.